dialoguekit/core/dialogue.py

```diff
@@ -34,6 +35,7 @@ def __init__(
         else:
             self._conversation_id = conversation_id
         self._utterances: List[Utterance] = []
+        self._utterance_feedbacks: List[UtteranceFeedback] = []
```

Comment:
Consider storing these in a dict instead.

Reply:
Done